database/src/main/scala/cromwell/database/slick/CallCachingSlickDatabase.scala

```diff
@@ -0,0 +1,53 @@
+package cromwell.database.slick
+
+import cromwell.database.sql.CallCachingStore
+import cromwell.database.sql.tables.{CallCachingHashEntry, CallCachingMetainfoEntry, CallCachingResultSimpletonEntry}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait CallCachingSlickDatabase extends CallCachingStore {
+  this: SlickDatabase =>
+
+  import dataAccess.driver.api._
+
+  override def addToCache(metainfo: CallCachingMetainfoEntry, hashes: Iterable[(String, String)], result: Iterable[(String, String, String)])(implicit ec: ExecutionContext): Future[Unit] = {
+    val insertMetainfo = dataAccess.callCachingMetainfoAutoInc += metainfo
+    runTransaction(insertMetainfo) map { metainfoId =>
+      val hashesToInsert = hashes map { case (hashKey, hashValue) => CallCachingHashEntry(hashKey, hashValue, metainfoId, None) }
+      val resultToInsert = result map { case (simpletonKey, simpletonValue, wdltype) => CallCachingResultSimpletonEntry(simpletonKey, simpletonValue, wdltype, metainfoId, None) }
+
+      val insertHashes = dataAccess.callCachingHashAutoInc ++= hashesToInsert
+      val insertResult = dataAccess.callCachingResultSimpletonAutoInc ++= resultToInsert
+
+      runTransaction(DBIO.sequence(List(insertMetainfo, insertHashes, insertResult)))
+    } map { _ => () }
+  }
+
+  override def ccmidsMatchingHashes(hashKeyValuePairs: Seq[(String, String)])(implicit ec: ExecutionContext): Future[Set[Int]] = {
+
+    val actions = hashKeyValuePairs map { case (hashKey, hashValue) => dataAccess.ccmidsForHashMatch(hashKey, hashValue).result }
+
+    def setIntersection(current: Set[Int], next: Seq[Int]) = current.intersect(next.toSet)
+
+    runTransaction(DBIO.sequence(actions)) map { _.toList } map {
+      case Nil => Set.empty
+      case head :: Nil => head.toSet
+      case head :: tail => tail.foldLeft(head.toSet)(setIntersection)
+    }
+  }
+
+  override def fetchReturnCodeAndResultSimpletons(ccmid: Int)(implicit ec: ExecutionContext): Future[Option[(Option[Int], Seq[CallCachingResultSimpletonEntry])]] = {
```

Comment:
Oh yeah I actually did make one, just in the engine layer. I can definitely move it down